rslib/src/ankihub/login.rs

```diff
@@ -0,0 +1,46 @@
+// Copyright: Ankitects Pty Ltd and contributors
+// License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html
+
+use reqwest::Client;
+use serde::Deserialize;
+use serde::Serialize;
+
+use crate::ankihub::http_client::HttpAnkiHubClient;
+use crate::prelude::*;
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct LoginRequest {
+    // FIXME: we need to pass either `username` or `email`
```

Comment:
The add-on uses a regex to determine whether the user entered an email or a username, but maybe it makes more sense to move the logic to the AnkiHub backend so clients don't have to worry about that.